Add media learning area. (#3215)
GoogleChrome/web.dev: src/site/content/en/media/accessibility/index.md

> ---

Reviewer: Def need to update this URL

> ---

Reviewer: Let's also mention this page under https://web.dev/accessible/#understand-semantics-and-basic-screen-reader-support …?

Author: Done.

layout: post
title: Accessibility with the <track> tag
authors:
  - joemedley
description: |
  Accessibility isn't like icing on a cake. It's never anything you put on a
  backlog with the hope of introducing later. Captions and screen reader
  descriptions are the only way many users can experience your videos, and in
  some jurisdictions, they're even required by law or regulation.
date: 2020-08-20

> updated: 2020-08-20

Reviewer: ```suggestion
updated: 2020-08-20
tags:
  - media
  - video
  - accessibility
```

---

Accessibility isn't like icing on a cake. It's never anything you put on a
backlog with the hope of introducing it later. Captions and screen reader
descriptions are the only way many users can experience your videos, and in some
jurisdictions, they're even required by law or regulation.

To add captions or screen reader descriptions to a web video, add a
[`<track>`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/track) tag
to a `<video>` tag. In addition to captions and screen reader descriptions, tags
may also be used for subtitles and chapter titles. The can also help search
engines understand what's in a video. Those capabilities are outside the scope
fo this article.

> <figure class="w-figure  w-figure--inline-right">

Reviewer: ```suggestion
<figure class="w-figure">
```
This looks nice but it's causing the code sample to get truncated

![image](https://user-images.githubusercontent.com/4713486/91112289-8c521200-e637-11ea-9cc7-7f029b37dc76.png)

>   <img src="./chrome-android-track-landscape-5x3.jpg" alt="Screenshot showing captions displayed using the track element in Chrome on Android">

Reviewer: ```suggestion
  <img class="w-screenshot w-screenshot--filled" src="./chrome-android-track-landscape-5x3.jpg" alt="Screenshot showing captions displayed using the track element in Chrome on Android">
```

</video>
```

The `<track>` tag is similar to the `<source>` element in that both have a `src`
attribute that points to referenced content. For a `<track>` tag, it points to a
[WebVTT file](https://developer.mozilla.org/en-US/docs/Web/API/WebVTT_API).  The
`label` attribute specifies the how a particular track will be identified in the
interface. To provide tracks for multiple languages add a separate `<track>` tag
for each WebVTT file you're providing and indicate the language using the
`srclang` attribute. The `default` attribute indicates which language track is
the default.

## Define captions in track file

Below is a hypothetical WebVTT file for the demo linked to above. The file is a text file containing a series of *cues*. Each cue is a block of text to display on screen and the time range during which it will be displayed.

```text
WEBVTT

00:00.000 --> 00:04.999
Man sitting on a tree branch, using a laptop.

00:05.000 --> 00:08.000
The branch breaks, and he starts to fall.

...
```

Each item in a track file is called a cue. Each cue has a start time and end
time separated by an arrow, with cue text in the line below. Cues can optionally

> also have IDs: 'railroad' and 'manuscript' in the example below. Cues are

Reviewer: ```suggestion
also have IDs: `railroad` and `manuscript` in the example below. Cues are
```

separated by an empty line.

```text

> WEBVTT FILE

Reviewer: Does the file literally start with `WEBVTT FILE`? Why does this one start with `WEBVTT FILE` but the previous one started with `WEBVTT`?

railroad
00:00:10.000 --> 00:00:12.500
Left uninspired by the crust of railroad earth

manuscript
00:00:13.200 --> 00:00:16.900
that touched the lead to the pages of your manuscript.
```

Cue times are in hours:minutes:seconds:milliseconds format. Parsing is strict.
Numbers must be zero padded if necessary: hours, minutes, and seconds must have
two digits (00 for a zero value) and milliseconds must have three digits (000

> for zero). There is an excellent WebVTT validator at quuz.org/webvtt, which

Reviewer: ```suggestion
for zero). There is an excellent WebVTT validator at [Live WebVTT Validator](https://quuz.org/webvtt/), which
```

checks for errors in time formatting, and problems such as non-sequential times.

You can create a VTT file by hand, thought there are [services that will create
them for

> you](https://www.google.com/search?sxsrf=ALeKk00bALi7XyBF8G5l7fK0eE1rwJEraQ%3A1596834624634&ei=QMMtX7G-JKm-0PEPlMCT0AI&q=webvtt+services&oq=webvtt+services&gs_lcp=CgZwc3ktYWIQAzIFCAAQzQIyBQgAEM0COgQIABBHUKZEWMxIYI1VaABwAXgAgAGAAYgBswKSAQMxLjKYAQCgAQGqAQdnd3Mtd2l6wAEB&sclient=psy-ab&ved=0ahUKEwix3sHEgIrrAhUpHzQIHRTgBCoQ4dUDCAw&uact=5).

Reviewer: ```suggestion
you](https://www.google.com/search?q=webvtt+services).
```

> ---

Author: @rkochman Kayce asked me make sure we covered everything in this article: https://www.html5rocks.com/en/tutorials/video/basics/

It had some accessibility material that I didn't have. I rewrote the section to include it. Please take a look.

Reviewer 2: LGTM. We're going to include the Javascript content in phase 2, correct?
